### PR TITLE
Bug 1254043 - [TV][Home] Remember cursor position for filter tabs

### PR DESCRIPTION
**Pull request by luke-chang**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3d0ae3c6285bc60039b8256f0d226c89748b3cfe

File: tv_apps/smart-home/js/home.js [hunk 1]
```diff
@@ -516,6 +515,20 @@
...
+        if (key == 'down') {
+          var dest = this.spatialNavigator.navigate(focus, key);
+          if (dest && dest.matches &&
+              dest.matches('#filter-tab-group smart-button')) {
+            var currentFilterElement = document.querySelector(
+              '#filter-tab-group smart-button[data-icon-type="' +
+              this.filterManager.getCurrentFilter() + '"]');
+            if (currentFilterElement) {
+              this.spatialNavigator.focus(currentFilterElement);
+              return;
```
Comment (rexboy7) [inline tv_apps/smart-home/js/home.js]: Is this return required?

File: tv_apps/smart-home/js/home.js [hunk 2]
```diff
@@ -516,6 +515,20 @@
       var focus = this.spatialNavigator.getFocusedElement();
 
       if (!(focus.CLASS_NAME == 'XScrollable' && focus.move(key))) {
```
Comment (rexboy7) [inline tv_apps/smart-home/js/home.js]: This is unrelated to your patch... but I think we can use
```
if (focus.CLASS_NAME == 'XScrollable' && focus.move(key)) {
  return;
}
```
then put your rest conditions to reduce indents.

File: tv_apps/smart-home/js/home.js (hunk 1 above)
Comment (rexboy7) [inline tv_apps/smart-home/js/home.js]: Oh, I figured it out. Just ignore my comment.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=39ec5c97df7d8acb0aa50b8e9f7b9fe8360b3471
